### PR TITLE
add support for FreeBSD, pet puppet-lint

### PR DESCRIPTION
**Pull request by fraenki**
This adds support for FreeBSD and handles the following differences:

* group `root` is named `wheel` on FreeBSD
* additional software is installed under `/usr/local`
* cachedir is `/usr/local/tarsnap-cache`
* `tarsnap-rotate.erb`: handle the absence of GNU coreutils on FreeBSD (and potential other platforms)

Besides that I've added `params.pp` to better handle parameters for different operating systems. Furthermore I've added some fixes to make puppet-lint happy.

### CODE REVIEW

File: templates/tarsnap-archive.erb [hunk 1]
```diff
@@ -3,4 +3,4 @@
 title="$1"
 shift
 
-<%= @path %> --quiet -c -f $title.$(date +%Y%m%d%H%M) $@
+<%= scope.lookupvar("tarsnap::path") %> --quiet -c -f $title.$(date +%Y%m%d%H%M) $@
```
Comment (igalic) [inline templates/tarsnap-archive.erb]: why is this necessary?

Comment (igalic) [inline templates/tarsnap-archive.erb]: you could just explicitly `include ::tarsnap` in `tarsnap::install`, that would pull path explicitly into the scope, making this unecessary.

Comment (fraenki) [inline templates/tarsnap-archive.erb]: @igalic Tried to explicitly `include ::tarsnap`, but it doesn't work for me. (Running Puppet master 3.7.4 and Puppet agent 3.6.2.)

Comment (igalic) [inline templates/tarsnap-archive.erb]: hm… i don't know… i should ping @hlindberg

**Comment by hlindberg** [comment]
An include of `::something` should work the same way in both future and current parsers. There has been issues around future parser and include. Make sure to test on 3.7.5 with future parser.

**Comment by fraenki** [comment]
It seems to be related to future parser, theforeman did something similar in https://github.com/theforeman/puppet-foreman_proxy/commit/1ea866780389dc92026944e995d40b2ed76bb2ad to pass future parser tests https://github.com/theforeman/puppet-foreman_proxy/pull/163.

**Comment by fraenki** [comment]
Did a quick test on a VM with puppet 3.7.5: With `parser = future` all variables in the erb templates are out-of-scope. However, when using the default parser it works as expected.

**Comment by hlindberg** [comment]
How are you getting the variables? You should use the @varname syntax.

**Comment by hlindberg** [comment]
I logged this ticket: https://tickets.puppetlabs.com/browse/PUP-4328, but as noted on the ticket I am not sure it really should be "fixed"

**Comment by fraenki** [comment]
@igalic I've updated my submission to handle the absence of GNU coreutils on FreeBSD. Do you have _strong_ objections against the use of `scope.lookupvar()`? Otherwise I'd suggest to keep it until the (future parser) bugs have been ironed out.

**Comment by igalic** [comment]
nope, i'm fine to go with what works. i'll log an issue to fix it in the future, @fraenki. thank you for now

**Comment by igalic** [comment]
@fraenki thanks again. i've now released this as [0.4.3](https://forge.puppetlabs.com/brainsware/tarsnap/0.4.3/readme)
